### PR TITLE
Ensure we handle duration, fn and scope correctly

### PR DESCRIPTION
**Pull request by marcjansen**
This ensures that we handle the already configurable keys `duration`, `scope` and `fn` of our `waitConf` correctly.

Please review.

### CODE REVIEW

File: src/view/MapLoadingStatusBar.js [hunk 1]
```diff
@@ -212,8 +215,31 @@ Ext.define('BasiGX.view.MapLoadingStatusBar', {
             if (!waitConf.text){
                 waitConf.text = me.lookupViewModel().get('loadingText');
             }
+            if (waitConf.duration) {
+                // when we have a duration set, we need to make sure that when
+                // The duration is exceeded, we internally reset our loading
```
Comment (chrismayer) [inline src/view/MapLoadingStatusBar.js]: Type `the`

File: src/view/MapLoadingStatusBar.js [hunk 2]
```diff
@@ -212,8 +215,31 @@ Ext.define('BasiGX.view.MapLoadingStatusBar', {
...
+                // counter. Since the user can also set a custom function to
+                // execute, we have to handle this case as well.
 
-            me.show();
+                // 1. Our handler when duration is reached.
+                var resetLoading = Ext.Function.bind(me.resetLoading, me);
+                if (waitConf.fn && Ext.isFunction(waitConf.fn)) {
+                    // 2. The user has set a handler when duration is reached
+                    //    We creeate a function that will first call the user
```
Comment (chrismayer) [inline src/view/MapLoadingStatusBar.js]: Typo `create`

File: src/view/MapLoadingStatusBar.js [hunk 3]
```diff
@@ -212,8 +215,31 @@ Ext.define('BasiGX.view.MapLoadingStatusBar', {
...
+                    //    function, than ours
```
Comment (hwbllmnn) [inline src/view/MapLoadingStatusBar.js]: Typo `then`

File: src/view/MapLoadingStatusBar.js (hunk 1 above)
Comment (marcjansen) [inline src/view/MapLoadingStatusBar.js]:
```suggestion
                // the duration is exceeded, we internally reset our loading
```

File: src/view/MapLoadingStatusBar.js (hunk 2 above)
Comment (marcjansen) [inline src/view/MapLoadingStatusBar.js]:
```suggestion
                    //    We create a function that will first call the user
```

File: src/view/MapLoadingStatusBar.js (hunk 3 above)
Comment (marcjansen) [inline src/view/MapLoadingStatusBar.js]:
```suggestion
                    //    function, then ours
```

**Pull request merged**